lib/Authen/SASL/Perl/OAUTHBEARER.pm

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2025 Aditya Garg <gargaditya08@live.com>
+# Copyright (c) 2025 Julian Swagemakers <julian@swagemakers.org>
+# All rights reserved.
+# This program is free software; you can redistribute it and/or
+# modify it under the same terms as Perl itself.
+
+package Authen::SASL::Perl::OAUTHBEARER;
+
+use strict;
+use vars qw(@ISA);
+
+@ISA     = qw(Authen::SASL::Perl);
+
+my %secflags = (
+    noanonymous => 1,
+);
+
+sub _order { 1 }
+
+sub _secflags {
+    shift;
+    scalar grep { $secflags{$_} } @_;
+}
+
+sub mechanism { 'OAUTHBEARER' }
+
+sub client_start {
+    my $self = shift;
+    $self->{stage} = 0;
+
+    # This will generate the oauthbearer string used for authentication.
+    #
+    # "n,a=" {User} ",^Ahost=" {Host} "^Aport=" {Port} "^Aauth=Bearer " {Access Token} "^A^A
+    #
+    # The first part `n,a=" {User} ",` is the gs2 header described in RFC5801.
+    # * gs2-cb-flag `n` -> client does not support CB
+    # * gs2-authzid `a=" {User} "`
+    #
+    # The second part are key value pairs containing host, port and auth as
+    # described in RFC7628.
+    #
+    # https://datatracker.ietf.org/doc/html/rfc5801
+    # https://datatracker.ietf.org/doc/html/rfc7628
+    my $username = $self->_call('user');
+    my $token    = $self->_call('pass'); # OAuth 2.0 access token
+    my $auth_string = "n,a=$username,\001host=$self->{host}\001port=$self->{port}\001auth=Bearer $token\001\001";
```

Comment:
Where does the value `$self->{port}` come from? The value of `$self->{host}` is populated from the `$host` parameter in `client_new`, but it seems that does not include local and remote ports.

From https://github.com/gbarr/perl-authen-sasl/blob/master/example_pl#L25-L28 I'm concluding the user of the SASL object should set the local and remote ports as properties, which means that their value will be available as `$self->property( 'localport' );` (if set under that name)

Reply:
We have to pass the $port variable by the email client

Reply:
I actually think we can drop host and port here. Reading the RFC7628[0] the values seem to be optional. Gmail is working fine without the values. Also looking at Net::SMTP[1] it's currently not supported.

[0] https://datatracker.ietf.org/doc/html/rfc7628#section-3.1
[1] https://github.com/steve-m-hay/perl-libnet/blob/35b9d4edb15a39f6c09694481ec3ac910936f085/lib/Net/SMTP.pm#L208

Reply:
Ah nice, I myself was wondering why my gmail account is not failing xD